3.7/ubuntu/Dockerfile

```diff
@@ -0,0 +1,271 @@
+RUN curl --verbose --head --fail --fail-early $OTP_SOURCE_URL 1>/dev/null && \
+    curl --verbose --location --silent --fail --fail-early --output $OTP_PATH.tar.gz $OTP_SOURCE_URL && \
+    echo "$OTP_SOURCE_SHA256  $OTP_PATH.tar.gz" > $OTP_PATH.sha256 && \
+    sha256sum --check --strict $OTP_PATH.sha256 && \
+    tar -xvf $OTP_PATH.tar.gz --directory $OTP_PATH --strip-components=1
+
+# Configure Erlang/OTP for compilation, disable unused features & applications
+# http://erlang.org/doc/applications.html
+# ERL_TOP is required for Erlang/OTP makefiles to find the absolute path for the installation
+RUN cd $OTP_PATH && \
+    export ERL_TOP=$OTP_PATH && \
+    ./otp_build autoconf && \
+    export CFLAGS="$(dpkg-buildflags --get CFLAGS)" && \
+    ./configure \
+      --host="$(dpkg-architecture --query DEB_HOST_GNU_TYPE)" \
+      --build="$(dpkg-architecture --query DEB_BUILD_GNU_TYPE)" \
+      --prefix=$OTP_INSTALL_DIR \
+      --disable-sctp \
+      --disable-silent-rules \
+      --disable-dynamic-ssl-lib \
+      --with-ssl=$OPENSSL_INSTALL_DIR \
+      --enable-clock-gettime \
+      --enable-hybrid-heap \
+      --enable-kernel-poll \
+      --enable-shared-zlib \
+      --enable-threads \
+      --enable-smp-support \
+      --enable-hipe \
+      --with-microstate-accounting=extra \
+      --without-odbc \
+      --without-diameter \
+      --without-erl_interface \
+      --without-ftp \
+      --without-jinterface \
+      --without-megaco \
+      --without-ssh \
+      --without-tftp \
+      --without-wx \
+      --without-debugger \
+      --without-dialyzer \
+      --without-et \
+      --without-observer \
+      --without-reltool \
+      --without-common_test \
+      --without-eunit \
+      --without-edoc \
+      --without-erl_docgen
+
+# Compile & install Erlang/OTP
+RUN cd $OTP_PATH && export ERL_TOP=$OTP_PATH && make -j "$(getconf _NPROCESSORS_ONLN)" GEN_OPT_FLGS="-O2 -fno-strict-aliasing" && make install
+
+# Delete all src dirs to keep the final image size down, they add an extra 19MB
+# TODO Could this be done better?
+RUN find $OTP_INSTALL_DIR/lib/erlang/lib -maxdepth 2 -mindepth 2 -name "src" -exec rm -fr {} \;
+
+# Record Erlang/OTP artefacts size
+RUN du -kh -d 1 $OTP_INSTALL_DIR/lib/erlang >> /OTP_BUILD_ARTEFACTS
+
+# Check that Erlang/OTP crypto & ssl were compiled against OpenSSL correctly
+RUN $OTP_INSTALL_DIR/bin/erl -noshell -eval 'io:format("~p~n~n~p~n~n", [crypto:supports(), ssl:versions()]), init:stop().'
+
+# Use the latest stable RabbitMQ release
+ARG RABBITMQ_VERSION="3.7.10"
+ARG RABBITMQ_SOURCE_URL="https://github.com/rabbitmq/rabbitmq-server/releases/download/v$RABBITMQ_VERSION/rabbitmq-server-generic-unix-$RABBITMQ_VERSION.tar.xz"
+ARG RABBITMQ_PATH=/usr/local/src/rabbitmq-$RABBITMQ_VERSION
+ARG RABBITMQ_INSTALL_DIR=/opt/rabbitmq
+RUN mkdir -p $RABBITMQ_INSTALL_DIR
+# https://www.rabbitmq.com/signatures.html#importing-gpg
+ENV RABBITMQ_PGP_KEY_ID="0x6B73A36E6026DFCA"
```

Comment:
This should be the full fingerprint (which will allow `gpg` to verify that the key it fetches is the key we expected), ala `0A9AF2115F4687BD29803A206B73A36E6026DFCA` or `0A9A F211 5F46 87BD 2980  3A20 6B73 A36E 6026 DFCA` with spaces if you want (`gpg` will accept either if you've got quotes, which the command below does). :+1: